src/components/ImageView/index.js

```diff
@@ -129,55 +137,20 @@ class ImageView extends PureComponent {
      * @param {Number} imageHeight
      */
     setImageRegion(imageWidth, imageHeight) {
-        let width = imageWidth;
-        let height = imageHeight;
-        const containerHeight = this.state.containerHeight;
-        const containerWidth = this.state.containerWidth;
-
-        // return if image not loaded yet
-        if (imageHeight <= 0 || containerHeight <= 0) {
+        if (imageHeight <= 0) {
             return;
         }
-
-        // Fit the image to container size if image small than container.
-        const aspectRatio = Math.min(containerHeight / imageHeight, containerWidth / imageWidth);
-        if (aspectRatio > 1) {
-            width *= (aspectRatio);
-            height *= (aspectRatio);
-        }
-        let imgLeft = (this.props.windowWidth - width) / 2;
-        let imgRight = ((this.props.windowWidth - width) / 2) + width;
-        let imgTop = (this.props.windowHeight - height) / 2;
-        let imgBottom = ((this.props.windowHeight - height) / 2) + height;
-        const isScreenWiderThanImage = (this.props.windowWidth / width) > 1;
-        const isScreenTallerThanImage = (this.props.windowHeight / height) > 1;
-        const aspect = width / height;
-        if (aspect > 1 && !isScreenWiderThanImage) {
-            // In case Width fit Screen width and Height not fit the Screen height
-            const fitRate = this.props.windowWidth / width;
-            imgLeft = 0;
-            imgRight = this.props.windowWidth;
-            imgTop = (this.props.windowHeight - (fitRate * height)) / 2;
-            imgBottom = imgTop + (fitRate * height);
-        } else if (aspect <= 1 && !isScreenTallerThanImage) {
-            // In case Height fit Screen height and Width not fit the Screen width
-            const fitRate = this.props.windowHeight / height;
-            imgTop = 0;
-            imgBottom = this.props.windowHeight;
-            imgLeft = (this.props.windowWidth - (fitRate * width)) / 2;
-            imgRight = imgLeft + (fitRate * width);
-        }
-
+        const containerHeight = this.state.containerHeight;
+        const containerWidth = this.state.containerWidth;
+        const width = imageWidth;
+        const height = imageHeight;
         const newZoomScale = Math.min(containerWidth / width, containerHeight / height);
-        this.setState({
+
+        this.setState(prevState => ({
             imgWidth: width,
-            zoomScale: newZoomScale,
             imgHeight: height,
-            imageLeft: imgLeft,
-            imageTop: imgTop,
-            imageRight: imgRight,
-            imageBottom: imgBottom,
-        });
+            zoomScale: containerHeight ? newZoomScale : prevState.zoomScale,
```

Comment:
I'm a little obsessed with this comment (https://github.com/Expensify/App/issues/8119#issuecomment-1101107165) and i found the reason.
Usually setImageRegion calls first and we dont need to set zoomScale we just set the imgWidth and imgHeight. But sometimes the call order can be change and setImageRegion function calls last so we have last chance to set zoomScale.  Here video and picture:

https://user-images.githubusercontent.com/6093207/166516719-e3606a29-3457-427d-8a30-12af61d65c9b.mov



![WhatsApp Image 2022-04-09 at 14 41 10](https://user-images.githubusercontent.com/6093207/162572604-0be6564e-0d63-489f-8bd9-c484cdf5366f.jpeg)

Reply:
i'm not sure how to reproduce but i don't mind keeping this. But let's add a comment explaining why its needed so that someone in future does not easily remove this by accident.